chore: update docs for new alerts and reporting feature

leocape opened the pull request:
### SUMMARY
Added docs for setting up and configuring the new alerts and reports in V.1.0

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
Using the templates outlined in the doc, run these steps to get the new alerts and reporting feature working. After these steps are completed, you should now be able to create and test alerts and reports

1. Create a new directory and create the Dockerfile
2. Build the extended image using the Dockerfile
3. Create the `docker-compose.yaml` file in the same directory
4. Create a new sub directory called `config`
5. Create the `superset_config.py` file in the `config` sub directory
6. Run the image using `docker-compose up` in the same directory as the `docker-compose.py` file

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

leocape committed:
update docs for new alerts and reporting feature

leocape committed:
fix punctuation

leocape committed:
fix punctuation

leocape committed:
fix punctuation

fix bullets
fix redis depends on name in docker-compose template

leocape committed:
fix bullets

Yann-OAF commented:
FYI, I have just sent PR #13116 which would enable this on Kubernetes by (optionally) running Celery beat as a separate singleton pod.

The rest of the configuration can be injected using the existing chart mechanism:
* Install webdrivers by overriding the worker's command (although doing this at startup time isn't great... the best would be to have them preinstalled in the image...)
* Pass all the necessary `superset_config.py` overrides from `values.yaml`

leocape committed:
Add extra steps for getting v1.0.1 running

step added: Upgrade the db
step added: init
step added: setup admin
fix redis superset naming
add . to docker build command

leocape committed:
fix worker to use gevent instead of prefork

change postgres to internal so it persists on docker-compose down

leocape committed:
add comments to superset_config.py

vnourdin commented on docs/src/pages/docs/installation/email_reports.mdx:
```diff
@@ -6,8 +6,309 @@ index: 10
 ---
 
-## Scheduling and Emailing Reports
+## Alerts and Reports
```
I'm not really comfy about putting the Alerts & Reports doc in a page named `email-reports`, as this doc concern alerts too, and not only emails but also Slack :shrug:

leocape replied:
I'd definitely agree, was really just for organisation so that in the menu there isn't 'Alerts and reporting' and then below it 'Scheduling and Emailing reports' which would be confusing imo, but if @srinify can help we could put it into its own section 'Alerts and Reporting', and then have the legacy reporting stuff remain at the bottom of this file?

vnourdin replied:
I agree, renaming and keep the old approach at the end seems the best!

Commit:
ref: Re-organize a bit and make it easier to read

Commit:
ref: rm useless level

nytai commented:
@leocape this is an excellent community effort in documenting a highly requested and very confusing feature set. I'm wondering how much of the config changes can be included in the default superset files (`superset/config.py`, `Dockerfile`, `docker-compose.yml`), so only minimal changes are required to enable this feature for testing. Documentation can then focus more on how to deploy this in prod-like environments (ie, not using docker-compose). Thoughts @dpgaspar? Are there any blockers (licensing) to adding the browser + webdriver to the Dockerfile?

nytai commented:
I also opened https://github.com/apache/superset/pull/13143 which includes a new docker-compose file for non-dev use cases. I added a beat container which I realized was missing in the regular docker-compose workflow.

leocape commented:
> @leocape this is an excellent community effort in documenting a highly requested and very confusing feature set. I'm wondering how much of the config changes can be included in the default superset files (`superset/config.py`, `Dockerfile`, `docker-compose.yml`), so only minimal changes are required to enable this feature for testing. Documentation can then focus more on how to deploy this in prod-like environments (ie, not using docker-compose). Thoughts @dpgaspar? Are there any blockers (licensing) to adding the browser + webdriver to the Dockerfile?

Thanks @nytai with all the help from everyone on the Slack threads. Yes would be great to simplify this doc so that it focuses more/only on the feature related config. I could cut out the superfluous docker-compose and dockerfile if you can include the webdriver (and postgres?) into the default dockerfile?

leocape commented:
> #13116

Thanks @Yann-OAF I could link to it from this doc where I've placed the link placeholder?

leocape commented on docs/src/pages/docs/installation/email_reports.mdx:
```diff
@@ -6,8 +6,305 @@ index: 10
+    mv chromedriver /usr/bin && \
+    apt autoremove -yqq --purge && \
+    apt clean && \
+    rm -f google-chrome-stable_current_amd64.deb chromedriver_linux64.zip
```
I couldn't get it working with geckodriver/firefox, no matter which version I tried, it wouldn't launch correctly when trying to take the screenshot... I suspect there is a missing config somewhere that is needed for it

vnourdin replied:
It works with `geckodriver` too, we can adapt the doc to propose both approaches.

Yann-OAF commented:
> > #13116
> 
> Thanks @Yann-OAF I could link to it from this doc where I've placed the link placeholder?

Yes, sounds good. These are probably too much to integrate to this doc, and won't be of interest to everyone (only k8s users) so linking is probably enough! I'm thinking I might start a dedicated doc on running in Kubernetes

dpgaspar reviewed (commented):
> @leocape this is an excellent community effort in documenting a highly requested and very confusing feature set. I'm wondering how much of the config changes can be included in the default superset files (`superset/config.py`, `Dockerfile`, `docker-compose.yml`), so only minimal changes are required to enable this feature for testing. Documentation can then focus more on how to deploy this in prod-like environments (ie, not using docker-compose). Thoughts @dpgaspar? Are there any blockers (licensing) to adding the browser + webdriver to the Dockerfile?

I think it does not make sense to release a dockerfile with an webdriver + browser installed, and that's where we need to exclude all non compliant ASF packages/software. So IMO we should keep the docker file lean and clean, and add hooks so that it's easy to add additional packages or software (on this case) 

btw, chrome driver license is 3-BSD, so it's "compatible" with ASF.

nytai commented:
Ok, in that case we should keep the Dockerfile changes. I'm planning to add the beat container in https://github.com/apache/superset/pull/13143 so that should take care of some the docker-compose changes, though it will need to be updated to reference the new images.

shemExelate commented on docs/src/pages/docs/installation/email_reports.mdx:
```diff
@@ -6,8 +6,309 @@ index: 10
+4. Create a new sub directory called `config`
+5. Create the `superset_config.py` file in the `config` sub directory
+6. Run the image using `docker-compose up` in the same directory as the `docker-compose.py` file
+7. In a new terminal window, upgrade the DB by running `docker exec -it superset-1.0.1-extended superset db upgrade`
```
is there a better way to run 
```
superset db upgrade
superset init
superset fab create-admin
```
as part of startup without entering the container?

nytai replied:
i think the `superset-init` service in the docker-compose workflow should take care of this.

Commit:
docs: md changes + more details + kubernetes

Commit:
docs: Rename to Alerts and Reports

Commit:
docs: Link to Kubernetes doc

Commit:
docs: details for docker-compose + minor md changes

Commit:
docs: fix lists indent

Commit:
docs: remove unused config

Commit:
docs: Put detailed config and dockerfile before specific stuff

Commit:
docs: clean Dockerfiles

aliJabra commented:
@vnourdin - thanks for the documentation! I'm running superset locally installed via pip (1.0.1). Is docker also a preferred way to run the 'backend' for generating alerts and reports? 
I've superset_config.py properly configured (i think), can see the alerts and reports, slack & email stmp tokens etc. set. I can create alerts and reports, but they never trigger.. 

I have a celery worker: 
`celery worker --app=superset.tasks.celery_app:app --pool=prefork -O fair -c 4 -E  --concurrency=500 &
`
and a beat process running in bg:
`celery beat --app=superset.tasks.celery_app:app --pidfile /tmp/celerybeat.pid --schedule /tmp/celerybeat-schedule &
`

I've used the default celery settings as per config.py

`class CeleryConfig:  # pylint: disable=too-few-public-methods
    BROKER_URL = "sqla+sqlite:///celerydb.sqlite"
    CELERY_IMPORTS = ("superset.sql_lab", "superset.tasks")
    CELERY_RESULT_BACKEND = "db+sqlite:///celery_results.sqlite"
    CELERYD_LOG_LEVEL = "DEBUG"
    CELERYD_PREFETCH_MULTIPLIER = 1
    CELERY_ACKS_LATE = True
    CELERY_ANNOTATIONS = {
        "sql_lab.get_sql_results": {"rate_limit": "100/s"},
        "email_reports.send": {
            "rate_limit": "1/s",
            "time_limit": 600,
            "soft_time_limit": 600,
            "ignore_result": True,
        },
    }
    CELERYBEAT_SCHEDULE = {
        "email_reports.schedule_hourly": {
            "task": "email_reports.schedule_hourly",
            "schedule": crontab(minute=1, hour="*"),
        },
        "reports.scheduler": {
            "task": "reports.scheduler",
            "schedule": crontab(minute="*", hour="*"),
        },
        "reports.prune_log": {
            "task": "reports.prune_log",
            "schedule": crontab(minute=0, hour=0),
        },
    }


CELERY_CONFIG = CeleryConfig`

srinify commented:
Hey @vnourdin @leocape I'm merging this as-is for now! The clean up process is quite involved, I like some things about the older commit and the latest commit and there's a lot of context to resolve. For now, it'd be nice to just have the info out there.

So I'm merging for now, and I'll do a second pass next week to clean up and organize better!

Kingflyinger commented:
`version: '3.6'
services:
  redis-superset:
    image: redis:6.0.9-buster
    restart: on-failure`
redis-superset should be replaced by redis,as well as superset_config.py.
or change the depends_on
